File: leetcode/src/PalindromeNumber.java
public class PalindromeNumber {
    public static void main(String[] args) {
        PalindromeNumber instance = new PalindromeNumber();
        System.out.println(instance.isPalindrome(121));
        System.out.println(instance.isPalindrome(-121));
        System.out.println(instance.isPalindrome(10));
        System.out.println(instance.isPalindrome(1));
    }
    public boolean isPalindrome(int x) {
        String str = String.valueOf(x);
        if (str.length() == 1) {
            return true;
        } else {
            for (int i = 0, n = str.length(); i < n; i++) {
                if (str.charAt(i) != str.charAt(n - 1 - i)) {
                    break;
                }
                if ((i == n - 1 - i) || (i + 1 == n - 1 - i)) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: leetcode/src/python/palindrome_number.py
class PalindromeNumber:
    def isPalindrome(self, x: int) -> bool:
        s = str(x)
        for i, c in enumerate(s):
            j = len(s) - 1 - i
            if i > j:
                break
            if c != s[j]:
                return False
        return True

if __name__ == '__main__':
    x = 1231
    p = PalindromeNumber()
    print(p.isPalindrome(x))